src/CamelKJSONUtils.ts

```diff
@@ -126,3 +126,30 @@ export function shareMessage(outputChannel: vscode.OutputChannel, msg:string) {
 		console.log('[' + msg + ']');
 	}
 }
+
+export function toKebabCase (str : string) {
```

Comment:
I really see this function to be a weak point for the future. if the naming pattern is modified, we need something to detect it.
I can imagine several possibilities:
- provide integration test with a running Camel-K instance and tests that are "passing through" this method
- have a test upstream which is forcing them to notify us if something is changing (might not be easy to negociate)
- ask for library able to do it
- have a test pulling the camel-K code to test that it has not changed.

Reply:
Added a test locally that mimics what the Camel-K test is doing upstream at https://github.com/apache/camel-k/blob/master/pkg/util/kubernetes/sanitize_test.go

I think this is the best we can do in the short term. We can investigate further options down the line.

Reply:
the added test is checking that vscode-camel-k will be compatible with the current implementation of camel-k. It won't detect if Camel-K is changing the implementation. We still have the weak point.
I really think that we should add a way to detect that the camel-k implementation has not changed.

Reply:
I have added a new test that is a brute force method of ensuring that sanitize.go hasn't changed upstream on us.

Reply:
seems a good way to detect potential desynchronization. (agree a bit bruteforce)